contrib/search_api_facetapi/search_api_facetapi.install

```diff
@@ -9,29 +9,82 @@
-  config_set('search_api_facetapi.settings', 'date_format_search_api_facetapi_' . FACETAPI_DATE_MINUTE, 'H:i');
-  config_set('search_api_facetapi.settings', 'date_format_search_api_facetapi_' . FACETAPI_DATE_SECOND, 'H:i:S');
+  _search_api_facetapi_set_date_formats();
 }
 
 /**
  * Implements hook_uninstall().
  */
 function search_api_facetapi_uninstall() {
-  // TODO This variable was probably removed in Backdrop without replacement.
-  variable_del('search_api_facets_search_ids');
-  // We have to use the literal values here, as the Facet API module could have
-  // already been disabled at this point.
-  config_clear('search_api_facetapi.settings', 'date_format_search_api_facetapi_YEAR');
-  config_clear('search_api_facetapi.settings', 'date_format_search_api_facetapi_MONTH');
-  config_clear('search_api_facetapi.settings', 'date_format_search_api_facetapi_DAY');
-  config_clear('search_api_facetapi.settings', 'date_format_search_api_facetapi_HOUR');
-  config_clear('search_api_facetapi.settings', 'date_format_search_api_facetapi_MINUTE');
-  config_clear('search_api_facetapi.settings', 'date_format_search_api_facetapi_SECOND');
+  state_del('search_api_facets_search_ids');
+  _search_api_facetapi_set_date_formats('unset');
 }
 
-// TODO The old hook_update_N functions cannot be applied to Backdrop.
-function search_api_facetapi_update_7101() { }
+/**
+ * Implements hook_update_last_removed().
+ */
+function search_api_facetapi_update_last_removed() {
+  return 7101;
+}
+
+/**
+ * Delete unnecessary config file and ensure date formats are set.
+ */
+function search_api_facetapi_update_1100() {
+  config('search_api_facetapi.settings')->delete();
+  _search_api_facetapi_set_date_formats();
+}
+
+/**
+ * Helper to ensure the custom date formats exist.
+ */
+function _search_api_facetapi_set_date_formats($action = 'set') {
+  include_once BACKDROP_ROOT . '/core/includes/update.inc';
+  $save_needed = FALSE;
+  $date_config = config('system.date');
+  $defaults = array(
+    'module' => 'search_api_facetapi',
+  );
+  $date_config_formats = $date_config->get('formats');
+  $date_formats = array(
+    FACETAPI_DATE_YEAR => array(
+      'label' => t('Search facets - Years'),
+      'pattern' => 'Y',
+    ) + $defaults,
+    FACETAPI_DATE_MONTH => array(
+      'label' => t('Search facets - Months'),
+      'pattern' => 'F Y',
+    ) + $defaults,
+    FACETAPI_DATE_DAY => array(
+      'label' => t('Search facets - Days'),
+      'pattern' => 'F j, Y',
+    ) + $defaults,
+    FACETAPI_DATE_HOUR => array(
+      'label' => t('Search facets - Hours'),
+      'pattern' => 'H:__',
+    ) + $defaults,
+    FACETAPI_DATE_MINUTE => array(
+      'label' => t('Search facets - Minutes'),
+      'pattern' => 'H:i',
+    ) + $defaults,
+    FACETAPI_DATE_SECOND => array(
+      'label' => t('Search facets - Seconds'),
+      'pattern' => 'H:i:s',
+    ) + $defaults,
+  );
+  foreach ($date_formats as $type => $format) {
+    $type = strtolower($type);
```

Comment:
Small deatil: I was wondering why bothering making the value of this constant lowercase - later one in `search_api_facetapi_format_timestamp()` you have to transform it again to lwoercase. I think it's fine for the key to have some uppercase letters, no?

Reply:
@argiepiano I initially left it as uppercase but if you try to edit the date formats to make customizations it gives an error that isn't fixable. The machine name is not editable (because it's already in use) but it won't validate. The machine name help text indicates:

> A unique machine-readable name. Can only contain lowercase letters, numbers, and underscores.

Reply:
Got it! Thanks for clarifying.